lib/awful/placement.lua

```diff
@@ -1422,11 +1422,33 @@ function placement.restore(d, args)
 
     if not memento then return false end
 
-    memento.screen = nil --TODO use it
+    local x, y = memento.x, memento.y
+
+    -- Some people consider that once moved to another screen, then
+    -- the memento needs to be upgraded. For now this is only true for
+    -- maximization until someone complains.
+    if memento.screen and memento.screen.valid and args.context == "maximize"
+      and d.screen and get_screen(memento.screen) ~= get_screen(d.screen) then
+        -- Use the absolute geometry as the memento also does
+        local oldsgeo = memento.screen.geometry
```

Comment:
Should the screen's geometry be saved inside the memento? That way this would also work if the original screen has since be removed.
Also, the screen's geometry could have changed, so the `x`,`y` saved in there need to be fixed up anyway (the `get_screen(memento.screen) ~= get_screen(d.screen) ` should be dropped), I think.